x-pack/plugins/alerting/server/rules_client/methods/aggregate.ts

```diff
@@ -6,90 +6,34 @@
  */
 
 import { KueryNode, nodeBuilder } from '@kbn/es-query';
-import { RawRule, RuleExecutionStatusValues, RuleLastRunOutcomeValues } from '../../types';
+import type { AggregationsAggregationContainer } from '@elastic/elasticsearch/lib/api/typesWithBodyKey';
 import { AlertingAuthorizationEntity } from '../../authorization';
 import { ruleAuditEvent, RuleAuditAction } from '../common/audit_events';
 import { buildKueryNodeFilter } from '../common';
 import { alertingAuthorizationFilterOpts } from '../common/constants';
 import { RulesClientContext } from '../types';
+import { RawRule, AggregateOptions } from '../../types';
+import { validateRuleAggregationFields } from '../lib/validate_rule_aggregation_fields';
 
-export interface AggregateOptions extends IndexType {
-  search?: string;
-  defaultSearchOperator?: 'AND' | 'OR';
-  searchFields?: string[];
-  hasReference?: {
-    type: string;
-    id: string;
-  };
-  filter?: string | KueryNode;
-  maxTags?: number;
+export interface AggregateParams {
+  options?: AggregateOptions;
+  aggs: Record<string, AggregationsAggregationContainer>;
 }
 
-interface IndexType {
-  [key: string]: unknown;
-}
-
-export interface AggregateResult {
-  alertExecutionStatus: { [status: string]: number };
-  ruleLastRunOutcome: { [status: string]: number };
-  ruleEnabledStatus?: { enabled: number; disabled: number };
-  ruleMutedStatus?: { muted: number; unmuted: number };
-  ruleSnoozedStatus?: { snoozed: number };
-  ruleTags?: string[];
-}
-
-export interface RuleAggregation {
-  status: {
-    buckets: Array<{
-      key: string;
-      doc_count: number;
-    }>;
-  };
-  outcome: {
-    buckets: Array<{
-      key: string;
-      doc_count: number;
-    }>;
-  };
-  muted: {
-    buckets: Array<{
-      key: number;
-      key_as_string: string;
-      doc_count: number;
-    }>;
-  };
-  enabled: {
-    buckets: Array<{
-      key: number;
-      key_as_string: string;
-      doc_count: number;
-    }>;
-  };
-  snoozed: {
-    count: {
-      doc_count: number;
-    };
-  };
-  tags: {
-    buckets: Array<{
-      key: string;
-      doc_count: number;
-    }>;
-  };
-}
-
-export async function aggregate(
+export async function aggregate<T extends Record<string, unknown>>(
   context: RulesClientContext,
-  {
-    options: { fields, filter, maxTags = 50, ...options } = {},
-  }: { options?: AggregateOptions } = {}
-): Promise<AggregateResult> {
+  params: AggregateParams
+): Promise<T | undefined> {
+  const { options = {}, aggs } = params;
+  const { filter, maxTags, page = 1, perPage = 0, ...restOptions } = options;
```

Comment:
Nit: `maxTags` is unused here so perhaps it would make sense to remove it from the `AggregateOptions` type, at lease for this function?